lib/cea/mp4_cea_parser.js

```diff
@@ -0,0 +1,229 @@
+
+          const parsedTRUN = shaka.util.Mp4BoxParsers.parseTRUN(
+              box.reader, box.version, box.flags);
+
+          sampleData = parsedTRUN.sampleData;
+        })
+
+        .fullBox('tfhd', (box) => {
+          goog.asserts.assert(
+              box.flags != null,
+              'TFHD is a full box and should have valid flags.');
+
+          const parsedTFHD = shaka.util.Mp4BoxParsers.parseTFHD(
+              box.reader, box.flags);
+
+          // If specified, defaultSampleDuration and defaultSampleSize
+          // override the ones specified in the TREX box
+          defaultSampleDuration = parsedTFHD.defaultSampleDuration
+            || this.defaultSampleDuration_;
+
+          defaultSampleSize = parsedTFHD.defaultSampleSize
+            || this.defaultSampleSize_;
+
+          const trackId = parsedTFHD.trackId;
+
+          // Get the timescale from the track Id
+          if (this.trackIdToTimescale_.has(trackId)) {
+            timescale = this.trackIdToTimescale_.get(trackId);
+          }
+        })
+
+        .fullBox('tfdt', (box) => {
+          goog.asserts.assert(
+              box.version != null,
+              'TFDT is a full box and should have a valid version.');
+
+          const parsedTFDT = shaka.util.Mp4BoxParsers.parseTFDT(
+              box.reader, box.version);
+
+          baseMediaDecodeTime = parsedTFDT.baseMediaDecodeTime;
+        })
+        .box('mdat', (box) => {
+          if (baseMediaDecodeTime === null) {
+            // This field should have been populated by
+            // the Base Media Decode time in the TFDT box
+            return;
+          }
+          this.parseMdat_(box.reader, baseMediaDecodeTime, timescale,
+              defaultSampleDuration, defaultSampleSize, sampleData, on708Data);
+        })
+        .parse(data, /* partialOkay= */ false);
+  }
+
+  /**
+   * Parse MDAT box.
+   * @param {!shaka.util.DataViewReader} reader
+   * @param {!number} time
+   * @param {!number} timescale
+   * @param {!number} defaultSampleDuration
+   * @param {!number} defaultSampleSize
+   * @param {!Array<shaka.util.ParsedTRUNSample>} sampleData
+   * @param {function(!BufferSource, number)} on708Data
+   * @private
+   */
+  parseMdat_(reader, time, timescale, defaultSampleDuration,
+      defaultSampleSize, sampleData, on708Data) {
+    let sampleIndex = 0;
+    let sampleSize = sampleData.length ? sampleData[0].sampleSize
+      || defaultSampleSize: defaultSampleSize;
+
+    while (reader.hasMoreData()) {
+      const naluSize = reader.readUint32();
+      const naluType = reader.readUint8() & 0x1F;
+      if (naluType == shaka.cea.CeaParser.NALU_TYPE_SEI) {
+        const timeOffset = sampleData.length > sampleIndex ?
+            (sampleData[sampleIndex].sampleCompositionTimeOffset || 0)/timescale
+             : 0;
+
+        const seiProcessingTime = time/timescale + timeOffset/timescale;
```

Comment:
How about `(time + timeOffset) / timescale`?  Doing the division only once might reduce rounding errors.

Also, I don't understand the name seiProcessingTime.  It sounds like something to do with how long processing took or when it took place, but it looks like it holds a presentation timestamp for the data we're processing.